runtime/cmake/platform.cmake

```diff
@@ -74,3 +74,29 @@ endif()
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1")
+
+# Hook add_library and add_executable so that we can add rules to strip debug info
+# on all targets.
+# Note: when defining a function 'foo()' any existing definition of 'foo' gets mapped to '_foo'
+
+function(add_library target)
+    # Call the real add_library
+    _add_library(${ARGV})
+
+    # We only need to split debug info on shared libraries
+    if(("SHARED" IN_LIST ARGV) AND (NOT "IMPORTED" IN_LIST ARGV))
+        omr_split_debug("${target}")
+    endif()
+endfunction()
+
+function(add_executable target)
+    # Call the real add_executable
+    _add_executable(${ARGV})
+
+    if(NOT (
+        ("ALIAS" IN_LIST ARGV) OR
+        ("IMPORTED" IN_LIST ARGV)
+    ))
+        omr_split_debug("${target}")
```

Comment:
Indenting is wrong here.

Reply:
I think the `if` condition should appear on one line (for consistency and to avoid the appearance of improper indenting).